Duplicati/Library/Main/Controller.cs

```diff
@@ -671,21 +671,18 @@ private void SetupCommonOptions(ISetCommonOptions result, ref string[] paths, re
             foreach (Library.Interface.IGenericModule m in DynamicLoader.GenericLoader.Modules)
                 m_options.LoadedModules.Add(new KeyValuePair<bool, Library.Interface.IGenericModule>(Array.IndexOf<string>(m_options.DisableModules, m.Key.ToLower()) < 0 && (m.LoadAsDefault || Array.IndexOf<string>(m_options.EnableModules, m.Key.ToLower()) >= 0), m));
 
-            var conopts = new Dictionary<string, string>(m_options.RawOptions);
+            // Make the filter read-n-write able in the generic modules
+            var pristinefilter = string.Join(System.IO.Path.PathSeparator.ToString(), FilterExpression.Serialize(filter));
+            m_options.RawOptions["filter"] = pristinefilter;
+            
             var qp = new Library.Utility.Uri(m_backend).QueryParameters;
             foreach(var k in qp.Keys)
-                conopts[(string)k] = qp[(string)k];
-
-            // Make the filter read-n-write able in the generic modules
-            var pristinefilter = conopts["filter"] = string.Join(System.IO.Path.PathSeparator.ToString(), FilterExpression.Serialize(filter));
+                m_options.RawOptions[(string)k] = qp[(string)k];

```

Comment:
Hmm OK, is there ever an instance where you would want an IConnectionModule to be able to modify the options dictionary that is passed to it in Configure()? Because the problem I'm seeing is that RunScript takes a reference to the options dictionary that is passed to it in the Configure() method and then modifies this dictionary based on the script output. The modifications that RunScript makes must be visible to the rest of the process, so they're stored in m_options. The previous code would pass a copy of the dictionary to IConnectionModule instances, which means if they were to modify the dictionary (in a similar way to RunScript) those changes are not reflected in m_options and would not be visible to the rest of the process.

I have updated the code with the assumption that there is no need for an IConnectionModule to change the options. I have brought back the conopts dictionary so that options supplied on the URL would only be visible to modules that implement IConnectionModule. I thought maybe the reason to prevent other modules from seeing these options is that they contain usernames and passwords (eg?user=x&pwd=y). However, that is visible in the remoteurl variable of Execute() and visible to batch scripts in DUPLICATI__REMOTEURL so I'm actually not sure. Is there some other reason for wanting to separate them? I don't see an issue with letting URL options override command line options _always_, or vice versa, and it would eliminate the extra dictionary in the code.

IConnectionModule marks the module as wanting to see the URL connection options, right? Is there a downside to having other modules see those options?

If you want IConnectionModule instances to be able to change m_options, I'll need to make some additional changes.